### PR TITLE
LPD-10211: Migrate from Travis-CI to GitHub Actions

### PR DESCRIPTION
**Pull request by tlsvda**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+      - name: Authenticate composer
+        env:
+          COMPOSER_GITHUB_TOKEN: ${{ secrets.COMPOSER_GITHUB_TOKEN }}
+        run: composer config --global --auth github-oauth.github.com "$COMPOSER_GITHUB_TOKEN"
+
```
Comment (christiaan) [inline .github/workflows/build.yaml:33]: I do not think this step is needed

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Create build
+        run: make dist
+        env:
+          LOCO_PROJECT_API_KEY: ${{ secrets.LOCO_PROJECT_API_KEY }}
+          LOCO_USER_INTERFACE_API_KEY: ${{ secrets.LOCO_USER_INTERFACE_API_KEY }}
```
Comment (christiaan) [inline .github/workflows/build.yaml:38]: As far as I know no localizations are used in this project?

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+
+      - name: Authenticate composer
+        env:
+          COMPOSER_GITHUB_TOKEN: ${{ secrets.COMPOSER_GITHUB_TOKEN }}
+        run: composer config --global --auth github-oauth.github.com "$COMPOSER_GITHUB_TOKEN"
+
```
Comment (christiaan) [inline .github/workflows/build.yaml:68]: I don't think this is needed

File: .gitignore [hunk 4]
```diff
@@ -3,3 +3,4 @@
 /build
 /.phpcs-cache
 /.phpunit.result.cache
+/.idea
```
Comment (christiaan) [inline .gitignore:6]: This is unrelated to this repository.

You should ignore the .idea directory on a global level using the git config option core.excludesfile. See http://egorsmirnov.me/2015/05/04/global-gitignore-file.html

File: .github/workflows/test.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+      - name: Authenticate composer
+        env:
+          COMPOSER_GITHUB_TOKEN: ${{ secrets.COMPOSER_GITHUB_TOKEN }}
+        run: composer config --global --auth github-oauth.github.com "$COMPOSER_GITHUB_TOKEN"
+
```
Comment (christiaan) [inline .github/workflows/test.yaml:31]: This is not needed.

File: composer.json [hunk 6]
```diff
@@ -15,6 +15,7 @@
     },
     "require-dev": {
         "behat/behat": "^3.4",
+        "dealerdirect/phpcodesniffer-composer-installer": "^0.6.2",
```
Comment (christiaan) [inline composer.json:18]: Why is this package suddenly needed?

Shouldn't the minimum version of `leviy/coding-standard` just be upped?

File: .github/workflows/build.yaml [hunk 7]
```diff
@@ -0,0 +1,71 @@
+name: Build
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (christiaan) [inline .github/workflows/build.yaml:6]: ubuntu-latest is soon going to move to ubuntu-20.04. When that happens `shivammathur/setup-php@v2` will not
be able to install php 7.2 anymore.

Comment (shivammathur) [reply .github/workflows/build.yaml:6]: @christiaan It will install PHP 7.2, will just take some time to install. 🙂

Comment (christiaan) [reply .github/workflows/build.yaml:6]: Thanks that is good to know 😄.

**Review by denniscoorn — APPROVED** [review]
LGTM

**Review by christiaan — CHANGES_REQUESTED** [review]
The Phar file still needs to be built and added to the release.

**Comment by denniscoorn** [comment]
> The Phar file still needs to be built and added to the release.

Oops, indeed. I've missed that part. Good call 👍

**Review by christiaan — APPROVED** [review]
